generic_config_updater/field_operation_validators.py

```diff
@@ -1,10 +1,119 @@
-from sonic_py_common import device_info
+import os
 import re
+import json
+import jsonpointer
+import subprocess
+from sonic_py_common import device_info
+from .gu_common import GenericConfigUpdaterError
 
-def rdma_config_update_validator():
-    version_info = device_info.get_sonic_version_info()
-    asic_type = version_info.get('asic_type')
 
-    if (asic_type != 'mellanox' and asic_type != 'broadcom' and asic_type != 'cisco-8000'):
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+GCU_TABLE_MOD_CONF_FILE = f"{SCRIPT_DIR}/gcu_field_operation_validators.conf.json"
+
+def get_asic_name():
+    asic = "unknown"
+    
+    if os.path.exists(GCU_TABLE_MOD_CONF_FILE):
+        with open(GCU_TABLE_MOD_CONF_FILE, "r") as s:
+            gcu_field_operation_conf = json.load(s)
+    else:
+        raise GenericConfigUpdaterError("GCU table modification validators config file not found")
+    
+    asic_mapping = gcu_field_operation_conf["helper_data"]["rdma_config_update_validator"]
+    
+    if device_info.get_sonic_version_info()['asic_type'] == 'cisco-8000':
+        asic = "cisco-8000"
+    elif device_info.get_sonic_version_info()['asic_type'] == 'vs':
+        asic = "td2"
```

Comment:
>asic [](http://example.com/codeflow?start=8&length=4)

This is not always true. Please check [https://github.com/sonic-net/sonic-buildimage/blob/220ea74cbb7e5cc783a27d7ca9f966899d81b7b3/platform/vs/docker-sonic-vs/Dockerfile.j2#L176](https://github.com/sonic-net/sonic-buildimage/blob/220ea74cbb7e5cc783a27d7ca9f966899d81b7b3/platform/vs/docker-sonic-vs/Dockerfile.j2#L176)

Reply:
Updated the PR to fully consider vs hwskus